### PR TITLE
Remove user inputs printed in logging and error statements

### PR DESCRIPTION
**Pull request by naveentatikonda**
Signed-off-by: Naveen Tatikonda <navtat@amazon.com>

### Description
As per the security recommendation, removing user inputs from being printed in logs and error statements

 
### Check List
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by VijayanB** [comment]
Is there any reason for removing it from Error message, since this will be displayed only to the user?

File: src/main/java/org/opensearch/knn/index/IndexUtil.java [hunk 1]
```diff
@@ -139,17 +139,16 @@ public static ValidationException validateKnnField(
 
             ModelMetadata modelMetadata = modelDao.getMetadata(modelId);
             if (modelMetadata == null) {
-                exception.addValidationError(String.format("Model \"%s\" for field \"%s\" does not exist.", modelId, field));
```
Comment (vamshin) [inline src/main/java/org/opensearch/knn/index/IndexUtil.java:142]: I think these messages are useful for customer to identify issues with a particular model, especially when they have multiple models.

**Review by vamshin — COMMENTED** [review]
I think all the Exception messages are valid according to me to provide more concise information for customers to debug issues. I think we should revert this PR.

**Comment by naveentatikonda** [comment]
As per our offline discussion, closing this PR.

**Pull request closed without merging**